runelite-mixins/src/main/java/net/runelite/mixins/RSSpritePixelsMixin.java

```diff
@@ -68,4 +68,77 @@ public void toBufferedImage(BufferedImage img)
 
 		img.setRGB(0, 0, width, height, transPixels, 0, width);
 	}
+
+	@Inject
+	@Override
+	public BufferedImage toBufferedOutline(int color)
+	{
+		BufferedImage img = new BufferedImage(getWidth(), getHeight(), BufferedImage.TYPE_INT_ARGB);
+
+		toBufferedOutline(img, color);
+
+		return img;
+	}
+
+	@Inject
+	@Override
+	public void toBufferedOutline(BufferedImage img, int color)
+	{
+		int width = getWidth();
+		int height = getHeight();
+
+		if (img.getWidth() != width || img.getHeight() != height)
+		{
+			throw new IllegalArgumentException("Image bounds do not match SpritePixels");
+		}
+
+		int[] pixels = getPixels();
+		int[] newPixels = new int[width * height];
+		int pixelIndex = 0;
+
+		for (int y = 0; y < height; ++y)
+		{
+			for (int x = 0; x < width; ++x)
+			{
+				int pixel = pixels[pixelIndex];
+				if (pixel == 16777215 || pixel == 0)
+				{
+					// W
+					if (x > 0 && pixels[pixelIndex - 1] != 0)
+					{
+						pixel = color;
+					}
+					// N
+					else if (y > 0 && pixels[pixelIndex - width] != 0)
+					{
+						pixel = color;
+					}
+					// E
+					else if (x < width - 1 && pixels[pixelIndex + 1] != 0)
+					{
+						pixel = color;
+					}
+					// S
+					else if (y < height - 1 && pixels[pixelIndex + width] != 0)
+					{
+						pixel = color;
+					}
+					newPixels[pixelIndex] = pixel;
+				}
+				pixelIndex++;
+			}
+		}
+
+		int[] transPixels = new int[newPixels.length];
+
+		for (int i = 0; i < newPixels.length; i++)
```

Comment:
can you not include this in the other loop? You're already looping over each pixel there.